lib/inspec/cli.rb

```diff
@@ -373,18 +377,44 @@ def run_command(opts)
   end
 end
 
+#=====================================================================#
+#                        Pre-Flight Code
+#=====================================================================#
+
+help_commands = ['-h', '--help', 'help']
+version_commands = ['-v', '--version', 'version']
+commands_exempt_from_license_check = help_commands + version_commands
+
+#---------------------------------------------------------------------#
+# EULA acceptance
+#---------------------------------------------------------------------#
+require 'license_acceptance/acceptor'
 begin
-  # Handle help commands
-  # This allows you to use any of the normal help commands after the normal args.
-  help_commands = ['-h', '--help', 'help']
-  (help_commands & ARGV).each do |cmd|
-    # move the help argument to one place behind the end for Thor to digest
-    if ARGV.size > 1
-      match = ARGV.delete(cmd)
-      ARGV.insert(-2, match)
-    end
+  if (commands_exempt_from_license_check & ARGV.map(&:downcase)).empty? &&  # Did they use a non-exempt command?
+     !ARGV.empty?                                                           # Did they supply at least one command?
+    LicenseAcceptance::Acceptor.check_and_persist('inspec', Inspec::VERSION)
```

Comment:
If your logger is initialized by this point you can pass it into the `Acceptor` to get debug/info logs. It would look like `LicenseAcceptance::Acceptor.check_and_persist('inspec', Inspec::VERSION, logger: Inspec::Log)`